src/java/org/apache/cassandra/db/commitlog/CommitLogSegment.java

```diff
@@ -374,7 +374,7 @@ void waitForFinalSync()
 
     void waitForSync(int position, Timer waitingOnCommit)
     {
-        while (lastSyncedOffset < position)
+        try (Timer.Context ignored2 = commitLog.metrics.totalWaitingOnCommit.time())
```

Comment:
super nit: why is this `ignored2`? shouldn't conflict with anything in this scope, could remove the `2` suffix